src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -64,11 +65,19 @@ public Saml2SecurityTokenHandler()
         /// <summary>
         /// Gets or set the <see cref="Saml2Serializer"/> that will be used to read and write a <see cref="Saml2SecurityToken"/>.
         /// </summary>
+        /// <exception cref="ArgumentNullException">'value' is null.</exception>
         public Saml2Serializer Serializer
         {
-            get;
-            set;
-        } = new Saml2Serializer();
+            get { return _serializer; }
+
+            set
+            {
+                if (value == null)
```

Comment:
use 'null condition operators' pattern, always use LogHelper to throw exceptions.
_serializer = value ?? LogHelper.LogArgumentNullException(nameof(value))